### PR TITLE
[release] prepare v0.66.0

### PR DESCRIPTION
**Pull request by codeboten**
This PR updates:
- core to v0.66.0
- versions.yaml to 0.66.0
- changelog
- Prepare contrib-base for version v0.66.0

Signed-off-by: Alex Boten <aboten@lightstep.com>

### CODE REVIEW

File: exporter/awsemfexporter/go.mod [hunk 1]
```diff
@@ -7,17 +7,17 @@ require (
...
+	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/resourcetotelemetry v0.66.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/opencensus v0.66.0
 	github.com/stretchr/testify v1.8.1
-	go.opentelemetry.io/collector v0.65.0
+	go.opentelemetry.io/collector v0.66.0
 	go.opentelemetry.io/collector/component v0.65.0
 	go.opentelemetry.io/collector/consumer v0.65.0
-	go.opentelemetry.io/collector/pdata v0.65.0
-	go.opentelemetry.io/collector/semconv v0.65.0
+	go.opentelemetry.io/collector/pdata v0.66.0
```
Comment (bogdandrutu) [inline exporter/awsemfexporter/go.mod:19]: Some are not updated, I think the update-otel script does not know about these modules.

Comment (codeboten) [reply exporter/awsemfexporter/go.mod:19]: let me fix this 👍🏻

Comment (bogdandrutu) [reply exporter/awsemfexporter/go.mod:19]: I will  :)

Comment (codeboten) [reply exporter/awsemfexporter/go.mod]: Updated the modules in this PR.

updated the script in a separate PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16458

**Review by bogdandrutu — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,16 @@
...
+## v0.66.0
+
+### 💡 Enhancements 💡
+
+- `hostmetricsreceiver`: Add a new optional metric `process.signals_pending` to the metrics scraped by the `process` scraper of the `hostmetrics` receiver. (#14084)
+- `splunkhecreceiver`: Add a healthcheck endpoint as part of the splunkhecreceiver. Just returns 200 for now if the receiver is running. (#15367)
+- `filterprocessor`: Add ability to filter spans, span events, metrics, datapoints, and logs via OTTL conditions (#16369)
+- `googlecloudspannerreceiver`: Configurably mask the PII in lock stats metrics. (#16343)
+- `interna/coreinternal`: Split internal/coreinternal/processor into a separate internal/filter module (#16410)
+
```
Comment (tigrannajaryan) [inline CHANGELOG.md:16]: Do we need a note that says 0.65.0 was a bad release?

**Comment by tigrannajaryan** [comment]
Needs retract messages for modules, like this https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16422/files

File: CHANGELOG.md (hunk 2 above)
Comment (codeboten) [reply CHANGELOG.md:16]: Good point, added a note

**Comment by bogdandrutu** [comment]
> Needs retract 0.65.0 messages for modules, like this https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16422/files

I don't know if you need them in this PR necessary, but good to add them here.

**Comment by dmitryax** [comment]
We also need "retract v0.65.0" in all the go mod files. Can be done with make for-all CMD="go mod edit -retract v0.65.0"

**Comment by tigrannajaryan** [comment]
> Needs `retract 0.65.0` messages for modules, like this https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16422/files

Command to add to to all .mod files if you want one:
```
find . -type f -name *.mod -exec bash -c 'echo "retract v0.65.0 // Broken release that did not correctly update components to v0.65.0" >> "{}"' \;
```

**Comment by tigrannajaryan** [comment]
> We also need "retract v0.65.0" in all the go mod files. Can be done with make for-all CMD="go mod edit -retract v0.65.0"

Nice, I didn't know go mod edit existed.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**